### PR TITLE
Fix psalm issues

### PR DESCRIPTION
**Pull request by jaapio**
- Setup of psaml, which can be added to a github action when merged ;-)
- Fixed all existing issues that I was able to fix without changing to much code.

### CODE REVIEW

File: src/shared/PharRegistry.php [hunk 1]
```diff
@@ -111,7 +111,11 @@ public function getByUsage(Filename $filename): Phar {
 
     public function removeUsage(Phar $phar, Filename $destination): void {
         $pharNode  = $this->getFirstMatchingPharNode($phar->getName(), $phar->getVersion());
-        $usageNode = $this->dbFile->query(\sprintf('//phive:usage[@destination="%s"]', $destination), $pharNode)->item(0);
+        $usageNode = $this->dbFile->query(\sprintf('//phive:usage[@destination="%s"]', (string)$destination), $pharNode)->item(0);
+        if (null === $usageNode) {
```
Comment (theseer) [inline src/shared/PharRegistry.php]: Can we avoid Joda? :-)

File: src/shared/cli/Runner.php [hunk 2]
```diff
@@ -86,7 +86,7 @@ public function run(): int {
     /**
      * @throws ErrorException
      */
-    public function errorHandler(int $code, string $message, string $file, int $line, array $context): void {
+    public function errorHandler(int $code, string $message, string $file, int $line, array $context): bool {
```
Comment (theseer) [inline src/shared/cli/Runner.php]: This is technically incorrect.

File: src/shared/cli/input/ConsoleInput.php [hunk 3]
```diff
@@ -6,9 +6,12 @@ class ConsoleInput implements Input {
     /** @var Output */
     private $output;
 
-    /** @var resource */
+    /** @var false|resource */
```
Comment (theseer) [inline src/shared/cli/input/ConsoleInput.php]: I do understand why this change probably makes psalm happy, but this property can never actually be false if the constructor succeeded...

Maybe we should change the flow to check for false before the assignment?

File: src/shared/cli/Runner.php (hunk 2 above)
Comment (jaapio) [reply src/shared/cli/Runner.php]: I know that from an implementation perspective this is incorrect. But according to the php docs a errorhandle method must return a `bool` https://www.php.net/manual/en/function.set-error-handler.php

> If the function returns FALSE then the normal error handler continues.

File: src/shared/cli/input/ConsoleInput.php (hunk 3 above)
Comment (jaapio) [reply src/shared/cli/input/ConsoleInput.php]: I checked the implementation here, I do agree with you that this should be a resource, but the default value could also be false. So this type hint is correct. Please notice that I added a new check in `\PharIo\Phive\Cli\ConsoleInput::confirm` to check if the `inputStream` is a valid resource. 
Technically `STDIN` can be a `resource` or `false` that's what you assign, I could move the check to the constructor?

Comment (theseer) [reply src/shared/cli/input/ConsoleInput.php]: Let's do that.

File: src/shared/cli/Runner.php (hunk 2 above)
Comment (theseer) [reply src/shared/cli/Runner.php]: Point taken.

**Comment by theseer** [comment]
Thanks so much already!

**Comment by theseer** [comment]
Thank you so much! Took a while to review but finally had a chance to check and merge.

**Comment by theseer** [comment]
Looks like now Scrutinizer is unhappy ;)

https://scrutinizer-ci.com/g/phar-io/phive/inspections/e80278a8-e4f8-44fc-bcde-71d8fde5b04c/issues/

Not sure how do address that best. Maybe we should just stop using it.

**Comment by jaapio** [comment]
No problem that it took a bit longer. Thanks for all you do!
